src/NHapi.Base/Util/MessageIterator.cs

```diff
@@ -1,584 +1,707 @@
+                    {
+                        NextRep(parentPos);
+                    }
+                    else
+                    {
+                        nextExists = NextPosition(parentPos, direction, makeNewSegmentIfNeeded);
+                    }
+                }
+                catch (HL7Exception e)
+                {
+                    throw new ApplicationException("HL7Exception arising from bad index: " + e.Message);
+                }
+            }
+            else
+            {
+                NewSegment(currPos.Parent, direction);
+            }
+
+            return nextExists;
+        }
+
+        /// <summary> An index of a child structure within a group, consisting of the name and rep of
+        /// of the child.
+        /// </summary>
+        public class Index
+        {
+            /// <summary>
+            /// The name.
+            /// </summary>
+            [Obsolete("This method has been replaced by 'Name'.")]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage(
+                "StyleCop.CSharp.NamingRules",
+                "SA1307:Accessible fields should begin with upper-case letter",
+                Justification = "As this is a public member, we will duplicate the method and mark this one as obsolete.")]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage(
+                "StyleCop.CSharp.MaintainabilityRules",
+                "SA1401:Fields should be private",
+                Justification = "As this is a public member, we will duplicate the method and mark this one as obsolete.")]
+            public string name;
+
+            /// <summary>
+            /// The repetition.
+            /// </summary>
+            [Obsolete("This method has been replaced by 'Rep'.")]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage(
+                "StyleCop.CSharp.NamingRules",
+                "SA1307:Accessible fields should begin with upper-case letter",
+                Justification = "As this is a public member, we will duplicate the method and mark this one as obsolete.")]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage(
+                "StyleCop.CSharp.MaintainabilityRules",
+                "SA1401:Fields should be private",
+                Justification = "As this is a public member, we will duplicate the method and mark this one as obsolete.")]
+            public int rep;
+
+            /// <summary>
+            /// The index.
+            /// </summary>
+            /// <param name="name">name.</param>
+            /// <param name="rep">repetition.</param>
+            public Index(string name, int rep)
+            {
+                Name = name;
+                Rep = rep;
+            }
+
+            /// <summary>
+            /// Gets or sets the name.
+            /// </summary>
+            public string Name

```

Comment:
kill fields and just user props?

Reply:
the fields are public so this would be a breaking change - I added the properties here to replace the fields, and have marked the fields as obsolete

Reply:
ok sounds good.